### PR TITLE
Feature/library

### LINKED ISSUE #2734: Library building
Library building can assign citizens to it, which will are unemployed and will level up their skills there. Each library level 1 citizen.

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #2734 

# Changes proposed in this pull request:
- Adds a library with 5 levels
- Each level can employ up to level*2 citizens
- Citizen level up their skills slowly by studying.

Review please

### CODE REVIEW

**Review by marchermans — DISMISSED** [review]

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 1]
```diff
@@ -53,6 +53,7 @@
     public static final String TAG_ORES                   = "tagOres";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_ORES doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 2]
```diff
@@ -53,6 +53,7 @@
...
     public static final String TAG_BLOCK                  = "block";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_BLOCK doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 3]
```diff
@@ -53,6 +53,7 @@
...
     public static final String TAG_POS                    = "pos";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Warnings:
- Constant name TAG_POS doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 4]
```diff
@@ -53,6 +53,7 @@
...
+    public static final String TAG_BOOKCASES              = "bookcase";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_BOOKCASES doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 5]
```diff
@@ -53,6 +53,7 @@
...
     public static final String TAG_BUILDING_MANAGER       = "buildingManager";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_BUILDING_MANAGER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 6]
```diff
@@ -53,6 +53,7 @@
...
     public static final String TAG_CITIZEN_MANAGER        = "citizenManager";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_CITIZEN_MANAGER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 7]
```diff
@@ -53,6 +53,7 @@
...
     public static final String TAG_STATS_MANAGER          = "statsManager";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_STATS_MANAGER doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 8]
```diff
@@ -312,6 +313,23 @@
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Warnings:
- Constant name TAG_COLONY_ID doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 9]
```diff
@@ -312,6 +313,23 @@
      */
     public static final String TAG_TEAM_COLOR = "teamcolor";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_TEAM_COLOR doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 10]
```diff
@@ -312,6 +313,23 @@
...
 
+    /**
+     * Citizen data Tags.
+     */
+    public static final String TAG_FEMALE                 = "female";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_FEMALE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 11]
```diff
@@ -312,6 +313,23 @@
...
+    public static final String TAG_TEXTURE                = "texture";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_TEXTURE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 12]
```diff
@@ -312,6 +313,23 @@
...
+    public static final String TAG_LEVEL                  = "level";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_LEVEL doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 13]
```diff
@@ -312,6 +313,23 @@
...
+    public static final String TAG_EXPERIENCE             = "experience";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_EXPERIENCE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 14]
```diff
@@ -312,6 +313,23 @@
...
+    public static final String TAG_HEALTH                 = "health";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_HEALTH doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 15]
```diff
@@ -312,6 +313,23 @@
...
+    public static final String TAG_MAX_HEALTH             = "maxHealth";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_MAX_HEALTH doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 16]
```diff
@@ -312,6 +313,23 @@
...
+    public static final String TAG_SKILLS                 = "skills";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_SKILLS doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 17]
```diff
@@ -312,6 +313,23 @@
...
+    public static final String TAG_SKILL_STRENGTH         = "strength";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_SKILL_STRENGTH doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 18]
```diff
@@ -312,6 +313,23 @@
...
+    public static final String TAG_SKILL_STAMINA          = "endurance";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_SKILL_STAMINA doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 19]
```diff
@@ -312,6 +313,23 @@
...
+    public static final String TAG_SKILL_SPEED            = "charisma";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_SKILL_SPEED doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 20]
```diff
@@ -312,6 +313,23 @@
...
+    public static final String TAG_SKILL_INTELLIGENCE     = "intelligence";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_SKILL_INTELLIGENCE doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 21]
```diff
@@ -312,6 +313,23 @@
...
+    public static final String TAG_SKILL_DEXTERITY        = "dexterity";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_SKILL_DEXTERITY doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

File: src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java [hunk 22]
```diff
@@ -312,6 +313,23 @@
...
+    public static final String TAG_SATURATION             = "saturation";
```
Comment (LDTTeam-Buildsystem) [inline src/api/java/com/minecolonies/api/util/constant/NbtTagConstants.java]: ### Errors:
- Field has no usages.

### Warnings:
- Constant name TAG_SATURATION doesn't match regex 'CONST_[A-Z][A-Z_\d]*'

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New  | Old  | Diff |
| ------ |:-----:|:----:|:----:|:----:|
| Total  | 39795 | 1608 | 1511 | +97  |
| Errors |  592  |  0   |  0   |  +0  |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=15306&tab=Inspection&buildTypeId=bt88)

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 39795 |  4  |  4  |  0   |
| Errors |  592  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=15456&tab=Inspection&buildTypeId=bt88)

**Pull request merged**